src/python/CRABInterface/RESTTask.py

```diff
@@ -374,3 +377,24 @@ def addddmreqid(self, **kwargs):
 
         return []
 
+    def addrucioasoinfo(self, **kwargs):
+        if 'workflow' not in kwargs or not kwargs['workflow']:
+            raise InvalidParameter("Task name not found in the input parameters")
+        if 'transfercontainer' not in kwargs or not kwargs['transfercontainer']:
+            raise InvalidParameter("Transfer container name not found in the input parameters")
+        if 'transferrule' not in kwargs or not kwargs['transferrule']:
+            raise InvalidParameter("Transfer container's rule id not found in the input parameters")
+        if 'publishrule' not in kwargs or not kwargs['publishrule']:
+            raise InvalidParameter("Transfer container's rule id not found in the input parameters")
+
+        workflow = kwargs['workflow']
+        authz_owner_match(self.api, [workflow], self.Task) #check that I am modifying my own workflow
```

Comment:
better use operator authz as well here. We may not have user's X509 certificate forever in the scheduler host. E.g. if we move to use token auth. in there.

Reply:
I think CRAB REST does not change when we use tokens. Simply scripts authenticates with CMSWEB FE using bearer token and FE will send same header as now which at that point will have some "CRAB service" identity and the Crab Operator role. In my understanding Dario was charged with making this happen, i.e. have all communication from schedd hosts to REST happen via that token. User proxy will only be used to send it to grid jobs for data access.
Of course we "could" keep using user X509 proxy for communication with REST as long as we have one (for sure until end of Run3), but the idea is to get experience with running a token-based service.
IN O(two years)(?) we may also do w/o the proxy in the grid jobs (once token-aware RSE's are deployed everywhere). At that point the only user of X509 will be FTS-ASO.

Reply:
yes.

Reply:
Ahhhhhhhhhh. I never though about that.
Thanks. I will fix.